build/build-info

```diff
@@ -34,7 +35,27 @@ echo_build_properties() {
   echo branch=$(cd "$git_path" && git rev-parse --abbrev-ref HEAD)
   echo date=$(date -u +%Y-%m-%dT%H:%M:%SZ)
   echo url=$(cd "$git_path" && git config --get remote.origin.url)
-  echo gpu_architectures=$(cuobjdump "$libcudf_path" 2>/dev/null | awk -F_ '/arch =/ {print $2}' | sort -n -u | tr '\n' ';')
+  gpu_architectures=$(
+    cuobjdump "$libcudf_path" | gawk '
+      match($0, /arch = sm_([0-9]+)/, tmp) {
+        arch[tmp[1]] = 1
+      }
+
+      END {
+        n = asorti(arch)
+        if (n == 0) {
+          print "ERROR: No fatbin ELF / PTX code sections found" > "/dev/stderr"
+          exit(1)
+        }
+        s = arch[1]
+        for (i = 2; i <= n; i++) {
+          s = s ";" arch[i]
+        }
+        printf s
```

Comment:
We don't want to try to interpret formatting characters here
```suggestion
        print s
```

Reply:
Originally, I was not using the concatenation, and used `printf` in the loop to print without a newline suffix.

Here the newline is also unnecessary but it gets stripped away by the output capture `$( ...)` anyways, so print is fine too.

There is no format interpretation unless there are further arguments to printf. If due to a bug the first argument contains a format-control letter it will error out

```
bash-4.2$ gawk 'BEGIN { str="75;%s;86"; printf str }'
gawk: cmd. line:1: fatal: not enough arguments to satisfy format string
        `75;%s;86'
            ^ ran out for this one
```